frontend/document/filemanagerdocument.lua

```diff
@@ -0,0 +1,85 @@
+local Document = require("document/document")
+local util = require("util")
+
+local FileManagerDocument = Document:new{
+    _document = false,
+    provider = "filemanagerdocument",
+    provider_name = "File Manager",
+    providers = {},
```

Comment:
This will need some thinking, unless addHandler takes care or don't inserting the same provider twice, or if the implementation provides a way for Plugins to deregister their handler on teardown.

(This is a static class members, while plugins will call init on each FileManager/ReaderUI instantiation, so, right now, you get a bunch of duplicates and pinned references).

Reply:
But the cached approach is complicated by the fact that if the plugin registers an instance reference instead of a module-local, this'll pin the instance object, too.

Reply:
And DocumentRegistry has the same design issue is that's it's very much designed only for static handlers.

Reply:
I think I solved it. Now providers is an array with names of providers.
Also we just register an extension once, when the first handler for the extension is added. More than one handler for a extension shouldn't reach documentRegistry. Instead we show a menu to let the user choose between actions.

Reply:
Yup, that looks better at a glance ;).

Docs should make it clear that `open_func` should point to a static/class method and not an instance one (which shouldn't be a problem given the design; and holds in the example), though ;).